pkg/flare/flare.go

```diff
@@ -0,0 +1,435 @@
+// redacts secrets, and writes them out to file.
+func (f *Flare) ProcessEnvVars(outPath, fileName string) (retErr error) {
+	outputFile, err := os.OpenFile(outPath+fileName, os.O_WRONLY|os.O_CREATE, FilePremissions)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if e := outputFile.Close(); retErr == nil {
+			retErr = e
+		}
+	}()
+	bw := bufio.NewWriter(outputFile)
+	defer func() {
+		if e := bw.Flush(); retErr == nil {
+			retErr = e
+		}
+	}()
+
+	err = f.processEnvVars(bw)
+	return err
+}
+
+func (f *Flare) processEnvVars(bw *bufio.Writer) error {
+	if bw == nil {
+		return ErrNoWriterProvided
+	}
+
+	for _, e := range os.Environ() {
+		for _, p := range f.envVarPrefixes {
+			if strings.HasPrefix(e, p) {
+				re, err := f.redactSecrets(e)
+				if err != nil {
+					return err
+				}
+				if _, err := bw.WriteString(fmt.Sprintf("%s\n", re)); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// ZipFolder zips the files created during the flare process into a single file
+// for sharing and writes to outputPath
+func (f *Flare) ZipFolder(inputPath, outputPath string) error {
+	fl, err := os.Create(outputPath)
+	if err != nil {
+		return err
+	}
+	defer fl.Close()
+
+	w := zip.NewWriter(fl)
+	defer w.Close()
+
+	walker := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		ex, err := os.Executable()
+		if err != nil {
+			return err
+		}
+		exPath := filepath.Dir(ex)
+		relPath, err := filepath.Rel(exPath, path)
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		f, err := w.Create(relPath)
+		if err != nil {
+			return err
+		}
+
+		_, err = io.Copy(f, file)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+	err = filepath.Walk(inputPath, walker)
+	return err
+}
+
+// HasLogFileOutput checks the outputs property under the logging property of the config
+// to make sure logs are configured to be written to file.
+func (f *Flare) HasLogFileOutput(outputs []string) bool {
+	return slices.ContainsFunc(outputs, func(e string) bool {
+		return e != "" && e != "-" && e != "="
+	})
+}
+
+// SupportedLogFormat checks that the string provided is one of the supported LogFormats
+func SupportedLogFormat(format string) bool {
+	v := LogFormat(format)
+	return v == LogFormatJSON || v == LogFormatPlainText
+}
+
+// ProcessLogFiles processes log files in inputPath. If the inputPath is a file, it processes the file.
+// If inputPath is a directory, all files in the directory will be individually processed.
+// Processed files are read line-by-line. Each line is filtered according to the date range,
+// sanitized from secrets, and is written to the file in outputPath.
+func (f *Flare) ProcessLogFiles(inputPath, outputPath string, startDate, endDate *time.Time) (retErr error) {
```

Comment:
That's a good point. I'm leaving this for last, since I want to consider it a bit.

Reply:
I agree that this could be an issue in space-constrained containers. However, the current Zip impl. zips the entire folder, incl. the log files, config files, and env vars file. This will be of value only if I create a zip file writer and then pass it along to all of the processor functions and close it only after all the files are written.

I'll give it a try...

Reply:
Done.